SqrMelon/animationgraph/curveview.py

```diff
@@ -540,6 +540,13 @@ def _drawCurves(self, painter: QPainter, rows: Iterable[int], start: float, end:
             prevPx = None
             x = max(start, curve[0].time())
             while x < min(end, curve[-1].time()):
+                evaluatedCurveData = curve.evaluateWithSnapAndKey(x, precision)
+                y = evaluatedCurveData[0]
+                if self.__newStyle == True:
+                    if self.__selection.isKeySelected(row, evaluatedCurveData[1]) or self.__selection.isKeySelected(row, evaluatedCurveData[2]):
+                        painter.setPen(self.__PENS['selected'])
+                    else:
+                        painter.setPen(self.__PENS[identifier] if identifier in self.__PENS else self.__PENS['single'])
                 for key in curve:
                     if abs(key.time() - x) < precision * 0.5:
```

Comment:
I have duplicated the snapping here because I did not understand it properly when I cherry picked it.
You can completely remove this section from the start of the for loop to the end of the else case.
Then the next lines would need to be:
```
# if they y was snapped, we must snap the x as well
if evaluatedCurveData[1] == evaluatedCurveData[2] != -1:
    p = QPointF(curve[evaluatedCurveData[1]].time(), y)
else:
    p = QPointF(x, y)
```
So that the 'p' variable exists before we call sceneToPixel.